src/SFML/Graphics/Text.cpp

```diff
@@ -346,10 +346,13 @@ void Text::draw(RenderTarget& target, RenderStates states) const
 ////////////////////////////////////////////////////////////
 void Text::ensureGeometryUpdate() const
 {
-    // Do nothing, if geometry has not changed
-    if (!m_geometryNeedUpdate)
+    // Do nothing, if geometry has not changed and the font texture has not changed
+    if (!m_geometryNeedUpdate && m_font->getTexture(m_characterSize).m_cacheId == m_fontTextureId)
         return;
 
+    // Save the current fonts texture id
+    m_fontTextureId = m_font->getTexture(m_characterSize).m_cacheId;
```

Comment:
Same as above.

Reply:
Just noticed that `m_fontTextureId` isn't initialized to a value prior to this line, meaning that in the comparison above, it would contain an undefined value.

Reply:
👍 fixed